app/controllers/CheckYourAnswersController.scala

```diff
@@ -113,7 +114,8 @@ class CheckYourAnswersController @Inject()(identify: IdentifierAction,
             .subscribe(request)
             .map { response =>
               val registrationType = answers.get(RegistrationTypePage).getOrElse(RegistrationType.ThirdParty)
-              SubscriptionDetails(response, request, registrationType, answers.get(BusinessTypePage))
+              val businessName = if (registrationType == PlatformOperator) getBusinessName(answers) else None
+              SubscriptionDetails(response, request, registrationType, answers.get(BusinessTypePage), businessName)
```

Comment:
You can create an apply method in the SubscriptionDetails and pass response, request and answers only. You can move the logic of creating the object there which will simplify the controller a bit and will provide better testability.